### PR TITLE
[WIP] Get LibWeb compiling on FreeBSD

### PR DESCRIPTION
**Pull request by BertalanD**
This does not *entirely* supersede #14369, as I'm only concerned with what's necessary to get Ladybird to run on FreeBSD.

See https://github.com/awesomekling/ladybird/issues/19

These commits should probably be squashed...

### CODE REVIEW

File: AK/StackInfo.cpp [hunk 1]
```diff
@@ -25,13 +28,23 @@ StackInfo::StackInfo()
         perror("get_stack_bounds");
         VERIFY_NOT_REACHED();
     }
-#elif defined(__linux__)
+#elif defined(__linux__) or defined(__FreeBSD__)
```
Comment (ADKaster) [inline AK/StackInfo.cpp:31]: Will this work on every BSD that's not macOS?

Comment (BertalanD) [reply AK/StackInfo.cpp:31]: I couldn't find a way to query thread attributes on OpenBSD. What other OSes should I check?

Comment (ADKaster) [reply AK/StackInfo.cpp:31]: that was the main one I was thinking of. I'm guessing all the FreeBSD derivatives and NetBSD will have similar APIs, but no users yet :)

**Comment by tristan957** [comment]
SourceHut has support for FreeBSD CI among other BSDs, fyi. SourceHut is pretty generous and will give open source communities who are approved access to its CI pipelines for free in order to test BSD support. You can probably get in contact with Drew DeVault, and see about getting access to this, so your work doesn't regress.

Neovim takes advantage of this, I believe.

**Pull request merged**